### PR TITLE
Verify HFS solution

### PR DESCRIPTION
**Pull request by yuchenpang**
I also add two command line arguments
1) -p, --precision to control the precision of bqp2hfs
2) -ss, --show-solution to print the solution to stderr

### CODE REVIEW

File: lns_hfs.py [hunk 1]
```diff
@@ -168,9 +168,79 @@ def main(args):
...
+
+def remove_tmp_file(filename):
+    print('INFO: removing file {}'.format(filename), file=sys.stderr)
+    try:
+        os.remove(filename)
+    except:
+        print('WARNING: removing file {} failed'.format(filename), file=sys.stderr)
+
+
+def verify_solution(tmp_hfs_file, tmp_sol_file, best_objective):
```
Comment (ccoffrin) [inline lns_hfs.py]: This is a good validation to make, but what if there is an inconsistency between the bqpjson file and the HFS data model (e.g. a difference in the objective function's energy due to finite precision?).  

In addition to this check I recommend that we take the HFS solution and then evaluate its value using the bqpjson data model.  Then the results in `BQP_DATA` should be consistent with the other solvers that use floating point arithmetic.

**Review by ccoffrin — COMMENTED** [review]
One suggestion.

**Comment by yuchenpang** [comment]
> One suggestion.

**Comment by yuchenpang** [comment]
> One suggestion.

I just added the code to use the energy evaluated in bqpjson whenever possible, but instead of checking the error in code, I only print it out in the log because I feel it's hard to decide what comparing tolerance is good in general.

**Review by ccoffrin — APPROVED** [review]
LGTM.  Added #10, but this a low priority at the moment.

**Comment by ccoffrin** [comment]
FYI @yuchenpang, made some updates here and merging.

**Pull request merged**